### PR TITLE
Unit test exception flows in elasticsearch ClientFacade.

### PR DESCRIPTION
**Pull request by fdlk**
Includes upgrade to mockito v2 to allow mocking a couple final methods in response objects.

#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] User documentation updated
- [ ] (If you have changed REST API interface) view-swagger.ftl updated
- [ ] Test plan template updated
- [ ] Clean commits

### CODE REVIEW

**Comment by dennishendriksen** [comment]
Travis build fails due to test errors:
```[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project molgenis-data-rest: There are test failures.```

**Comment by dennishendriksen** [comment]
Mockito upgrade causes many deprecation warnings such as:
```org.mockito.Matchers in org.mockito has been deprecated```
We have an almost warning-free build, would be nice to keep it that way. I suggest fixing the deprecation warnings before upgrading Mockito.

File: molgenis-data-elasticsearch/src/main/java/org/molgenis/data/elasticsearch/client/ClientFacade.java [hunk 1]
```diff
@@ -522,6 +522,7 @@ public Explanation explain(SearchHit searchHit, QueryBuilder query)
 		}
 
 		String indexName = searchHit.getIndex();
+		//FIXME: ClientFacade shouldn't assume that typename equals typename
```
Comment (tommydeboer) [inline molgenis-data-elasticsearch/src/main/java/org/molgenis/data/elasticsearch/client/ClientFacade.java]: Did you add the FIXME's and TODO's in this file to the indexing document on Google Drive?

File: molgenis-data-elasticsearch/src/test/java/org/molgenis/data/elasticsearch/client/ClientFacadeTest.java [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+/**
+ * Unit tests for the exception flows of the ClientFacade.
+ * We assume the other flows are sufficiently covered by the integration tests.
+ */
+public class ClientFacadeTest
```
Comment (tommydeboer) [inline molgenis-data-elasticsearch/src/test/java/org/molgenis/data/elasticsearch/client/ClientFacadeTest.java]: You can extend from ```AbstractMockitoTest``` so you don't need to reset and initialize your mocks.

File: molgenis-data-elasticsearch/src/main/java/org/molgenis/data/elasticsearch/client/ClientFacade.java (hunk 1 above)
Comment (fdlk) [inline molgenis-data-elasticsearch/src/main/java/org/molgenis/data/elasticsearch/client/ClientFacade.java]: Now I have :) Thanks

**Pull request merged**